src/Evolution/Particles/MonteCarlo/InverseJacobianInertialToFluidCompute.cpp

```diff
@@ -24,20 +24,21 @@ void InverseJacobianInertialToFluidCompute::function(
         (spatial_velocity.get(d) - shift.get(d) / get(lapse));
   }
 
-  // Then, the other members of the tetrad are constructed using Gram-Shmidt
+  // Then, the other members of the tetrad are constructed using Gram-Schmidt
 
   // Temporary memory allocation
-  DataVector temp_dot_product = get(lapse) * 0.0;
-  std::array<DataVector, 4> tetrad_vector{temp_dot_product, temp_dot_product,
-                                          temp_dot_product, temp_dot_product};
-  tnsr::aa<DataVector,3,Frame::Inertial> spacetime_metric =
-    gr::spacetime_metric(lapse, shift, spatial_metric);
+  auto temp_dot_product = make_with_value<DataVector>(lapse, 0.0);
+  tnsr::aa<DataVector, 3, Frame::Inertial> spacetime_metric =
+      gr::spacetime_metric(lapse, shift, spatial_metric);
 
   // d = 1,2,3 for tetrad components built from x,y,z
   for (size_t d = 1; d < 4; d++) {
     // Base vector for Gram-Shmidt
     for (size_t i = 0; i < 4; i++) {
-      gsl::at(tetrad_vector, i) = (i == d ? 1.0 : 0.0);
+      inv_jacobian->get(i, d) = make_with_value<DataVector>(lapse, 0.0);
```

Comment:
Do `set_number_of_grid_points(&inv_jacobian->get(i, d), lapse)` (from `Utilities/SetNumberOfGridPoints.hpp`) and then `= 0.0` (or the `?:` expression you had before).  This avoids a DataVector allocation.  Might need a `make_not_null`.